onnxruntime/core/providers/cuda/nn/conv.h

```diff
@@ -206,6 +207,7 @@ class Conv : public CudaKernel {
   constexpr static auto kDefaultConvAlgo = CUDNN_CONVOLUTION_FWD_ALGO_IMPLICIT_PRECOMP_GEMM;
   static const cudnnConvolutionFwdAlgo_t kAllAlgos[];
   std::unique_ptr<Tensor> W_;
+  bool contrib_nhwc_;
```

Comment:
Isn't `NhwcConv` also a contrib op ? If so, can we choose a better naming of this var to differentiate the two kernels and leave a detailed comment for anyone else reading through this code later ?

Reply:
So it's basically the same as the new NHWC op, but the prepack is done in python transformers. I can add that as comment if you want to !

Reply:
Maybe rename as is_nhwc_domain_.

For NhwcConv (it is in ms domain), we used python script to transpose inputs, outputs and weights to NHWC layout during graph fusion. So weight transpose is not needed in inference.

Please add a comment. That could help reader

Reply:
Ok done.  Thanks for the recommendation.

Reply:
I don't think it's quite the same (at least not currently) as the NhwcConv has a different layout for the weights as they're also transposed. 

IIRC perf testing was pending to figure out which approach was best but I never heard the outcome of that. 

It would be relatively easy to update the Conv in the internal NHWC domain to be consistent (only a few EPs utilize it) so we don't have 2 internal operators doing the same thing (i.e. handling NHWC Conv). As long as the transpose optimizer can handle changing the weight layout to remove the transpose it would be pretty much the same as with PrePack (weight transpose is done during model initialization and not during Run) just with less code.

Reply:
Spoke offline with Scott - we will find a way to not have 2 operators doing the same operation (i.e.) NHWC Conv in the future.